### DISCUSSION TITLE
Is configuration for custom linters (plugins) possible via .golangci.yaml?

### DISCUSSION THREAD

**Discussion by george-e-shaw-iv** [golangci/golangci-lint#2116] [Q&A]
I've written a few custom linters and they're in the correct format, ready to be turned into a plugin and consumed by golangci-lint. My question is whether or not it's possible to pass flags to these custom linters via the `.golangci.yaml` file, similar to the way you'd configure `errcheck`, or any of the other built-in linters. I really need the ability to do this :/

**Comment by jaredallard** [comment]
I think this ultimately comes down to, is there a way to read config as a `golangci-lint` plugin? Correct me if I'm wrong @george-e-shaw-iv

**Comment by Antonboom** [comment]
On golangci-lint side you can transform `.golangci.yaml` config into flags for your linter.

Example:

1. My linter accepts [the flag](https://github.com/Antonboom/nilnil/blob/803825e11168b50aed97ec11b32249037863e3c9/pkg/analyzer/analyzer.go#L28)
2. But on `golangci-lint` side it's [a config](https://github.com/golangci/golangci-lint/blob/59f1c6e455a6b288d859662a54707e523a05989d/pkg/golinters/nilnil.go#L16).
3. `golangci-lint` [transforms](https://github.com/golangci/golangci-lint/blob/59f1c6e455a6b288d859662a54707e523a05989d/pkg/golinters/goanalysis/linter.go#L105) config map into the linter flags.

**Reply by Antonboom** [reply]
@george-e-shaw-iv, sorry, my bad

your question is about linters as **plugins**?
I don't see any code related to specific plugin configuration.

(cc) @ldez
